### PR TITLE
Remove unused vanilla code (closed #1658)

### LINKED ISSUE #1658: vanilla code
can we delete all the commented out vanilla code and delete even more unused stuff?

### PR DESCRIPTION
**Pull request by ChillerDragon**
Removed vanilla code from gamecontext and gamecontroller. I didn't search for the code all over the place I just remember seing it there a lot. Also removed some empty lines. Wanted to do this for a long time :)

thought there is an reason it is there. But we have git and the vanilla repo so I don't see any point keeping it there.

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -1120,8 +1078,6 @@ void CGameContext::OnClientConnected(int ClientID)
 	//players[client_id].init(client_id);
 	//players[client_id].client_id = client_id;
 
-	//(void)m_pController->CheckTeamBalance();
```
Comment (heinrich5991) [inline src/game/server/gamecontext.cpp]: Just a random drive-by comment: There's some more commented stuff above.

Comment (ChillerDragon) [reply src/game/server/gamecontext.cpp]: ya wasnt sure about that stuff. Do you think I should remove it as well?
